personalized_nlp/experiments/studemo_single_ann_tune.py

```diff
@@ -0,0 +1,132 @@
+        )
+
+        for model_type, embedding_dim, dp_emb, fold_num, lr_rate in product(
+                model_types, embedding_dims, dp_embs, range(fold_nums), lr_rates):
+            hparams = {
+                "dataset": type(data_module).__name__,
+                "model_type": model_type,
+                "embeddings_type": embeddings_type,
+                "embedding_size": embedding_dim,
+                "fold_num": fold_num,
+                "regression": regression,
+                "words_per_texts": words_per_text,
+                "min_word_count": min_word_count,
+                "dp_emb": dp_emb,
+                "nr_frozen_epochs": nr_frozen_epochs,
+                "weight_decay": weight_decay,
+                "learning_rate": lr_rate,
+                "max_seq_len": 256
+
+            }
+
+            logger = pl_loggers.WandbLogger(
+                save_dir=LOGS_DIR,
+                config=hparams,
+                project=wandb_project_name,
+                log_model=False,
+            )
+
+            output_dim = len(data_module.class_dims) if regression else sum(
+                data_module.class_dims)
+            text_embedding_dim = data_module.text_embedding_dim
+            model_cls = models_dict[model_type]
+
+            model = model_cls(output_dim=output_dim,
+                              text_embedding_dim=text_embedding_dim,
+                              word_num=data_module.words_number,
+                              annotator_num=data_module.annotators_number,
+                              dp=0.0,
+                              dp_emb=dp_emb,
+                              embedding_dim=embedding_dim,
+                              hidden_dim=100,
+                                nr_frozen_epochs=nr_frozen_epochs,
+                                embedding_type=embeddings_type,
+                                frozen = frozen,
+                              bias_vector_length=len(data_module.class_dims))
+
+            custom_callbacks = [SetWeightDecay(lr=lr_rate, weight_decay=weight_decay)]
+            if frozen == False:
+                custom_callbacks += [TransformerLrScheduler(warmup_proportion=0.1)]
+
+            train_test(data_module,
+                       model,
+                       epochs=epochs,
+                       lr=lr_rate,
+                       regression=regression,
+                       use_cuda=use_cuda,
+                       hparams=hparams,
+                       logger=logger,
+                       output_dir="./results",
+                       per_device_train_batch_size=batch_size,
+                       per_device_eval_batch_size=batch_size,
+                       evaluation_strategy=eval_strat,
+                         nr_frozen_epochs=nr_frozen_epochs,
+                       weight_decay=weight_decay,
+                       custom_callbacks=custom_callbacks)
+
+            logger.experiment.finish()
+            
```

Comment:
Github stills shows me that there is no new line, probably, there are some whitespaces which should be removed.